components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2676,6 +2677,72 @@ public static String replaceEmailDomain(String input) {
         return input;
     }
 
+    /**
+     * This method is used to get the APIProvider instance for a given username
+     *
+     * @param context API Context of the API
+     * @throws APIManagementException If the context is not valid
+     */
+    public static void validateAPIContext(String context) throws APIManagementException {
+        Pattern pattern = Pattern.compile(contextRegex);
+
+        if (context == null || context.isEmpty()) {
+            log.error("Invalid Context: Context cannot be empty or null");
+            throw new APIManagementException("Invalid Context: Context cannot be empty or null");
+        }
+
+        if (context.endsWith("/")) {
```

Comment:
shouldn't we silently handle these?

Reply:
We can remove this validation from here as this is already getting handled by the backend. But in `4.1.0` that error message is not a proper one and string replacement does not happen accurately.